pkg/blobstore/readfallback/read_fallback_blob_access.go~

```diff
@@ -77,13 +81,21 @@ func (eh *readFallbackErrorHandler) OnError(observedErr error) (buffer.Buffer, e
 		return nil, util.StatusWrap(observedErr, "Secondary")
 	}
 	if eh.secondary == nil {
-		// Both backends returned NOT_FOUND. Don't bother
-		// prepending the name of the backend.
+		// No secondary exists, so just return the error
 		return nil, observedErr
 	}
-	ba := eh.secondary
+
+	if eh.replicator != nil {
```

Comment:
The Buildbarn codebase tends to stick to the [null object pattern](https://en.wikipedia.org/wiki/Null_object_pattern) as much as possible.

Please add a NoopBlobReplicator, for which ReplicateMultiple() does nothing, and ReplicateSingle() does nothing more than call Get() on the source. The advantage of doing that is that we then have a universal mechanism for this, not specific to ReadFallbackBlobAccess.

Reply:
Done